curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/653639d63a45a077333312c8.md

```diff
@@ -7,23 +7,27 @@ dashedName: step-32
 
 # --description--
 
-Before playing the song, you need to make sure it starts from the beginning. This can be achieved by the use of the `currentTime` property of the `audio` object. 
+Before playing the song, you need to make sure it starts from the beginning. This can be achieved by the use of the `currentTime` property on the `audio` object. 
 
-Add an `if` statement to check whether the `userData?.currentSong` is `null` or if `userData?.currentSong.id` is strictly not equal `song.id`. Inside `if` block, set the `currentTime` property of the `audio` object to `0`.
+Add an `if` statement to check whether the `userData?.currentSong` is falsy *OR* if `userData?.currentSong.id` is strictly not equal to `song.id`. This condition will check if no current song is playing or if the current song is different from the one that is about to be played.
 
+Inside your `if` block, set the `audio.currentTime` property to `0`.
 
 # --hints--
 
-You should create an `if` statement with the condition `userData?.currentSong === null || userData?.currentSong.id !== song.id`.
+You should create an `if` statement that checks whether `userData.currentSong` is either falsy *OR* `userData.currentSong` is not equal to `song.id`. Don't forget to use optional chaining.
 
 ```js
-assert.match(code, /if\s*\(\s*userData\?\.currentSong\s*===\s*null\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id\s*\)\s*\{\s*/)
+assert.match(code, /if\s*\(\s*(?:!userData\?\.currentSong|\s*userData\?\.currentSong\s*===\s*null|\s*userData\?\.currentSong\s*===\s*undefined)\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id\s*|\s*userData\?\.currentSong\.id\s*!==\s*song\.id\s*\|\|\s*(?:!userData\?\.currentSong|\s*userData\?\.currentSong\s*===\s*null|\s*userData\?\.currentSong\s*===\s*undefined)\s*\)\s*\{\s*/);
```

Comment:
I've seen the issue and read the conversations, I've come up with a solution that satisfies all condiitons,

```suggestion
assert.match(code, /if\s*\(\s*(?:userData\?\.currentSong\s*===\s*(?:null|undefined)\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id|userData\?\.currentSong\.id\s*!==\s*song\.id\s*\|\|\s*userData\?\.currentSong\s*===\s*(?:null|undefined))\s*\)\s*\{\s*/);
```

look here, there's 3 blocks mainly 
first condition
```
userData\?\.currentSong\s*===\s*(?:null|undefined)
```
the or statement
```
\s*\|\|\s*
```
last condition
```
 userData\?\.currentSong\.id\s*!==\s*song\.id
```

I've written and simplified the regex in a way that it targets both cases of "Either this or that". try this solution @aaqib605